tests/dhcp_relay/test_dhcp_relay.py

```diff
@@ -96,8 +96,21 @@ def dut_dhcp_relay_data(duthost, ptfhost, testbed):
 
     return dhcp_relay_data_list
 
+@pytest.fixture(scope="module")
+def validate_dut_routes_exist(duthost, dut_dhcp_relay_data):
+    """Fixture to valid a route to each DHCP server exist
+    """
+    dhcp_servers = set()
+    for dhcp_relay in dut_dhcp_relay_data:
+        dhcp_servers |= set(dhcp_relay['downlink_vlan_iface']['dhcp_server_addrs'])
+
+    for dhcp_server in dhcp_servers:
+        result = duthost.shell("ip route get {0}".format(dhcp_server))
+        assert result["rc"] == 0 and "{0} via".format(dhcp_server) in result["stdout"], \
```

Comment:
Is `assert` working properly now, or should we use `pytesl.fail()` or Ying's new `pytest_assert`?

Reply:
Yes, I had no issues with assert. I have seen the assert with no locals variable message being triggered.

Reply:
here is a sample output of the assert (please note condition inversion in order to trigger assert)
```
duthost = <common.devices.SonicHost object at 0x7fcae13f5490>
dut_dhcp_relay_data = [{'client_iface': {'alias': u'fortyGigE0/52', 'name': u'Ethernet52', 'port_idx': 13}, 'downlink_vlan_iface': {'addr': ...['PortChannel0002', 'PortChannel0001', 'PortChannel0004', 'PortChannel0003'], 'uplink_port_indices': [29, 28, 31, 30]}]

    @pytest.fixture(scope="module")
    def validate_dut_routes_exist(duthost, dut_dhcp_relay_data):
        """Fixture to valid a route to each DHCP server exist
        """
        dhcp_servers = set()
        for dhcp_relay in dut_dhcp_relay_data:
            dhcp_servers |= set(dhcp_relay['downlink_vlan_iface']['dhcp_server_addrs'])
    
        for dhcp_server in dhcp_servers:
            result = duthost.shell("ip route get {0}".format(dhcp_server))
>           assert result["rc"] != 0 and "{0} via".format(dhcp_server) in result["stdout"], \
                "Failed to find route to DHCP server '{0}' with error '{1}".format(dhcp_server, result["stderr"])
E           AssertionError: Failed to find route to DHCP server '192.0.0.4' with error '
E           assert (0 != 0)

dhcp_relay/test_dhcp_relay.py:109: AssertionError
```